src/goto-instrument/contracts/contracts.cpp

```diff
@@ -375,24 +392,26 @@ void code_contractst::add_quantified_variable(
   }
 }
 
-void code_contractst::replace_old_parameter(
+void code_contractst::replace_history_parameter(
   exprt &expr,
   std::map<exprt, exprt> &parameter2history,
   source_locationt location,
   const irep_idt &mode,
-  goto_programt &history)
+  goto_programt &history,
+  const irep_idt &id)
 {
   for(auto &op : expr.operands())
   {
-    replace_old_parameter(op, parameter2history, location, mode, history);
+    replace_history_parameter(
+      op, parameter2history, location, mode, history, id);
   }
 
-  if(expr.id() == ID_old)
+  if(expr.id() == ID_old || expr.id() == ID_loop_entry)
   {
-    DATA_INVARIANT(
-      expr.operands().size() == 1, CPROVER_PREFIX "old must have one operand");
+    const auto &parameter = to_history_expr(expr, id).expression();
 
-    const auto &parameter = to_old_expr(expr).expression();
+    DATA_INVARIANT(
+      expr.operands().size() == 1, "A history variable must have one operand");
```

Comment:
The 2 lines above are redundant. Please remove those.

The number of operands check would be performed during `to_history_expr` call @ line 411, as I mentioned in https://github.com/diffblue/cbmc/pull/6278#discussion_r684686555.